From 3353a7e2a381075d3294536b50d5d715da1cc167 Mon Sep 17 00:00:00 2001
From: Ben Langmuir <blangmuir@apple.com>
Date: Tue, 23 Jul 2019 10:28:58 -0700
Subject: [PATCH] [presets] Add clang install to bots that test
 indexstore-db/sourcekit-lsp

At a minimum, we need to install libIndexStore to enable testing the
indexer, but it's better to also include clang/clangd, since this lets
us enable all of our tests on these bots. This commit uses the
swift-install-components and llvm-install-components from the Package
bots, which increases the similarity between our PR tests and package
bots.

In addition to the sourcekit-lsp/indexstore-db PR test bots, this
affects the following bots that test indexstore-db and didn't already
install clang:

* Linux
  * incremental
  * incremental long test
  * swiftpm PR test

* macOS
  * PR test
  * PR smoke test
  * incremental
  * swiftpm PR test

Based on looking at recent build logs for the package bots, this should
only add <5 seconds to the incremental build.
---
 utils/build-presets.ini | 52 +++++++++++------------------------------
 1 file changed, 13 insertions(+), 39 deletions(-)

diff --git b/utils/build-presets.ini a/utils/build-presets.ini
index fd09a779e4..9ec1df70af 100644
--- b/utils/build-presets.ini
+++ a/utils/build-presets.ini
@@ -18,9 +18,6 @@ dash-dash
 
 swift-install-components=compiler;clang-builtin-headers;stdlib;sdk-overlay;parser-lib;editor-integration;tools;toolchain-tools;testsuite-tools;sourcekit-xpc-service;swift-remote-mirror;swift-remote-mirror-headers;
 
-[preset: mixin_buildbot_install_components_with_clang]
-swift-install-components=compiler;clang-resource-dir-symlink;stdlib;sdk-overlay;parser-lib;toolchain-tools;license;sourcekit-xpc-service;swift-remote-mirror;swift-remote-mirror-headers
-llvm-install-components=llvm-cov;llvm-profdata;IndexStore;clang;clang-headers;compiler-rt;clangd
 
 [preset: mixin_buildbot_trunk_base]
 # Build standard library and SDK overlay for iOS device and simulator.
@@ -325,9 +322,7 @@ tvos
 watchos
 
 [preset: buildbot_incremental,tools=RA,stdlib=RA]
-mixin-preset=
-    buildbot_incremental_base_all_platforms
-    mixin_buildbot_install_components_with_clang
+mixin-preset=buildbot_incremental_base_all_platforms
 
 build-subdir=buildbot_incremental
 
@@ -356,7 +351,6 @@ skip-test-watchos
 install-swift
 install-llbuild
 install-swiftpm
-install-libcxx
 
 [preset: buildbot_incremental,tools=RA,stdlib=RA,xcode]
 mixin-preset=buildbot_incremental,tools=RA,stdlib=RA
@@ -524,7 +518,6 @@ swift-stdlib-build-type=RelWithDebInfo
 mixin-preset=
     buildbot_incremental_base
     lldb-smoketest,tools=RA
-    mixin_buildbot_install_components_with_clang
 build-subdir=buildbot_incremental
 
 # We build release+asserts.
@@ -558,7 +551,6 @@ sourcekit-lsp
 install-swift
 install-llbuild
 install-swiftpm
-install-libcxx
 
 # We need to build the unittest extras so we can test
 build-swift-stdlib-unittest-extra
@@ -725,14 +717,8 @@ swift-enable-ast-verifier=0
 #===------------------------------------------------------------------------===#
 # Linux Builders
 #===------------------------------------------------------------------------===#
-[preset: mixin_linux_install_components_with_clang]
-swift-install-components=autolink-driver;compiler;clang-resource-dir-symlink;stdlib;swift-remote-mirror;sdk-overlay;parser-lib;toolchain-tools;license;sourcekit-inproc
-llvm-install-components=llvm-cov;llvm-profdata;IndexStore;clang;clang-headers;compiler-rt;clangd
-
 [preset: mixin_linux_installation]
-mixin-preset=
-    mixin_lightweight_assertions
-    mixin_linux_install_components_with_clang
+mixin-preset=mixin_lightweight_assertions
 
 llbuild
 swiftpm
@@ -750,6 +736,8 @@ install-swiftpm
 install-xctest
 install-libicu
 install-prefix=/usr
+swift-install-components=autolink-driver;compiler;clang-resource-dir-symlink;stdlib;swift-remote-mirror;sdk-overlay;parser-lib;toolchain-tools;license;sourcekit-inproc
+llvm-install-components=llvm-cov;llvm-profdata;IndexStore;clang;clang-headers;compiler-rt;clangd
 install-libcxx
 build-swift-static-stdlib
 build-swift-static-sdk-overlay
@@ -980,9 +968,7 @@ build-ninja
 reconfigure
 
 [preset: buildbot_incremental_linux]
-mixin-preset=
-    buildbot_incremental_linux_base
-    mixin_linux_install_components_with_clang
+mixin-preset=buildbot_incremental_linux_base
 build-subdir=buildbot_incremental
 
 libicu
@@ -1002,7 +988,6 @@ install-swiftpm
 install-foundation
 install-libdispatch
 install-xctest
-install-libcxx
 
 [preset: buildbot_incremental_linux,long_test]
 mixin-preset=buildbot_incremental_linux
@@ -1079,7 +1064,6 @@ sourcekit-lsp=0
 # OS X Package Builders
 #===------------------------------------------------------------------------===#
 [preset: mixin_osx_package_base]
-mixin-preset=mixin_buildbot_install_components_with_clang
 ios
 tvos
 watchos
@@ -1122,8 +1108,6 @@ skip-install-swiftsyntax-module
 install-skstresstester
 install-swiftevolve
 install-playgroundsupport
-install-libcxx
-install-sourcekit-lsp
 
 install-destdir=%(install_destdir)s
 
@@ -1142,6 +1125,10 @@ test-installable-package
 # If someone uses this for incremental builds, force reconfiguration.
 reconfigure
 
+swift-install-components=compiler;clang-resource-dir-symlink;stdlib;sdk-overlay;parser-lib;toolchain-tools;license;sourcekit-xpc-service;swift-remote-mirror;swift-remote-mirror-headers
+llvm-install-components=llvm-cov;llvm-profdata;IndexStore;clang;clang-headers;compiler-rt;clangd
+install-libcxx
+
 # Path to the .tar.gz package we would create.
 installable-package=%(installable_package)s
 
@@ -1317,7 +1304,6 @@ swift-stdlib-build-type=Release
 mixin-preset=
     buildbot_incremental_base_all_platforms
     lldb-smoketest,tools=RA
-    mixin_buildbot_install_components_with_clang
 
 build-subdir=buildbot_incremental
 
@@ -1336,7 +1322,6 @@ sourcekit-lsp
 install-swift
 install-llbuild
 install-swiftpm
-install-libcxx
 
 # Build Playground support
 playgroundsupport
@@ -1383,9 +1368,7 @@ skip-test-osx
 #===------------------------------------------------------------------------===#
 
 [preset: buildbot_swiftpm_macos_platform,tools=RA,stdlib=RA]
-mixin-preset=
-    buildbot_incremental_base
-    mixin_buildbot_install_components_with_clang
+mixin-preset=buildbot_incremental_base
 
 build-subdir=buildbot_incremental
 
@@ -1402,7 +1385,6 @@ sourcekit-lsp
 install-swift
 install-llbuild
 install-swiftpm
-install-libcxx
 
 dash-dash
 
@@ -1415,9 +1397,7 @@ skip-test-llbuild
 #===------------------------------------------------------------------------===#
 
 [preset: buildbot_swiftpm_linux_platform,tools=RA,stdlib=RA]
-mixin-preset=
-    buildbot_incremental_base
-    mixin_linux_install_components_with_clang
+mixin-preset=buildbot_incremental_base
 
 build-subdir=buildbot_incremental
 
@@ -1439,7 +1419,6 @@ install-swiftpm
 install-foundation
 install-libdispatch
 install-xctest
-install-libcxx
 
 skip-test-swift
 skip-test-cmark
@@ -1490,7 +1469,6 @@ sourcekit-lsp
 install-swift
 install-llbuild
 install-swiftpm
-install-libcxx
 
 skip-test-swift
 skip-test-cmark
@@ -1498,9 +1476,7 @@ skip-test-llbuild
 skip-test-swiftpm
 
 [preset: buildbot_swiftpm_package_macos]
-mixin-preset=
-    buildbot_swiftpm_package_base
-    mixin_buildbot_install_components_with_clang
+mixin-preset=buildbot_swiftpm_package_base
 
 # Build stdlib for all platforms.
 ios
@@ -1508,9 +1484,7 @@ tvos
 watchos
 
 [preset: buildbot_swiftpm_package_linux]
-mixin-preset=
-    buildbot_swiftpm_package_base
-    mixin_linux_install_components_with_clang
+mixin-preset=buildbot_swiftpm_package_base
 
 libdispatch
 foundation
-- 
2.23.0.11

